tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -177,30 +177,9 @@ build_decompression_map(PlannerInfo *root, DecompressChunkPath *path, List *scan
 			/*
 			 * Normal column, not a metadata column.
 			 */
-			AttrNumber hypertable_attno = get_attnum(path->info->ht_rte->relid, column_name);
 			AttrNumber chunk_attno = get_attnum(path->info->chunk_rte->relid, column_name);
```

Comment:
Could we add an `Assert(compression_info->algo_id != _INVALID_COMPRESSION_ALGORITHM)` here?

Reply:
Oh after rereading your suggestion, this will not work all segmentby columns would fail that check since they have invalid compression algorihtm since they are not stored compressed.